### PR TITLE
Embed initial ramdisk inside the kernel binary

### PR DESCRIPTION
**Pull request by j-piecuch**
* Remove associated emulator options from the launcher script
* Minor refactoring -- see commit messages

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,16 +83,6 @@ format:
 test: mimiker.elf
 	./run_tests.py
 
-# Detecting whether initrd.cpio requires rebuilding is tricky, because even if
```
Comment (cahirwpz) [inline Makefile]: What was the rationale behind moving `initrd` stuff to subdirectory? Isn't that an unnecessary change?

File: mips/malta.c [hunk 2]
```diff
@@ -125,30 +125,15 @@ extern intptr_t parse_rd_start(const char *s);
...
-   */
-  assert(is_aligned(rd_start, PAGESIZE));
-  assert(is_aligned(rd_start + rd_size, PAGESIZE));
-  assert(rd_start == 0 || (intptr_t)__kernel_end <= rd_start);
-
-  intptr_t real_kernel_end =
-    (rd_start == 0) ? (intptr_t)(__kernel_end) : (intptr_t)(rd_start + rd_size);
-
-  pm_seg_t *seg = (pm_seg_t *)real_kernel_end;
+  pm_seg_t *seg = (pm_seg_t *)__kernel_end;
```
Comment (cahirwpz) [inline mips/malta.c]: I understand you'll change that to use `kernel_sbrk` with next PR. Is that correct?

Comment (j-piecuch) [reply mips/malta.c]: That's correct.

File: Makefile (hunk 1 above)
Comment (j-piecuch) [reply Makefile]: Since the source files are already separated into different directories (i.e. `mips`, `sys`, etc.), I thought I'd follow this pattern. I also didn't want to create intermediate files (`initrd.o`, `initrd.cpio`) in the root directory of the project.

Comment (cahirwpz) [reply Makefile]: Hmm... `initrd` directory stores only `Makefile` at the moment and it's unlikely it will contain anything else in foreseeable future. I'll revert that change. We should discuss such decisions next time.

**Comment by cahirwpz** [comment]
I'm aware of problems with `signal_segfault` test spuriously failing. Nevertheless I'll integrate this change with master.
